### PR TITLE
Update EIP-4844: Use EIP-6493 signature scheme

### PR DESCRIPTION
**Pull request by etan-status**

### CODE REVIEW

**Comment by g11tech** [comment]
I think it would be better to decouple review status change with the PR changes into 4844

File: EIPS/eip-4844.md [hunk 1]
```diff
@@ -161,25 +162,29 @@ The execution layer verifies the wrapper validity against the inner `Transaction
...
 
-
-The signature is verified and `tx.origin` is calculated as follows:
+The hashes for signature verification and for determining the public transaction identifier are computed according to the [EIP-6493](./eip-6493.md) transaction signature scheme:
 
 ```python
-def unsigned_tx_hash(tx: SignedBlobTransaction) -> Bytes32:
-    # The pre-image is prefixed with the transaction-type to avoid hash collisions with other tx hashers and types
-    return keccak256(BLOB_TX_TYPE + ssz.serialize(tx.message))
+def compute_blob_sig_hash(message: BlobTransaction) -> Root:
```
Comment (g11tech) [inline EIPS/eip-4844.md:168]:
```suggestion
def compute_sig_hash(message: BlobTransaction) -> Root:
```
context blob is clear so may be we can skip it

File: EIPS/eip-4844.md [hunk 2]
```diff
@@ -161,25 +162,29 @@ The execution layer verifies the wrapper validity against the inner `Transaction
...
+        GENESIS_HASH,
+        CHAIN_ID,
+    )
+    return compute_signing_root(message, domain)
 
-def get_origin(tx: SignedBlobTransaction) -> Address:
-    sig = tx.signature
-    # v = int(y_parity) + 27, same as EIP-1559
-    return ecrecover(unsigned_tx_hash(tx), int(sig.y_parity)+27, sig.r, sig.s)
+def compute_blob_tx_hash(signed_tx: SignedBlobTransaction) -> Root:
```
Comment (g11tech) [inline EIPS/eip-4844.md:177]:
```suggestion
def tx_hash(signed_tx: SignedBlobTransaction) -> Root:
```
signed_tx_hash can now be referenced without ambiguity to be tx_hash

File: EIPS/eip-4844.md [hunk 3]
```diff
@@ -161,25 +162,29 @@ The execution layer verifies the wrapper validity against the inner `Transaction
...
-The hash of a signed blob transaction should be computed as:
+`tx.origin` is calculated as follows:
 
 ```python
-def signed_tx_hash(tx: SignedBlobTransaction) -> Bytes32:
-    return keccak256(BLOB_TX_TYPE + ssz.serialize(tx))
+def get_origin(tx: SignedBlobTransaction) -> Address:
+    sig = tx.signature
+    # v = int(y_parity) + 27, same as EIP-1559
+    return ecrecover(compute_blob_sig_hash(tx), int(sig.y_parity)+27, sig.r, sig.s)
```
Comment (g11tech) [inline EIPS/eip-4844.md:187]:
```suggestion
    return ecrecover(compute_sig_hash(tx), int(sig.y_parity)+27, sig.r, sig.s)
```

File: EIPS/eip-4844.md [hunk 4]
```diff
@@ -161,25 +162,29 @@ The execution layer verifies the wrapper validity against the inner `Transaction
 - The KZG commitments match the blob contents. (Note: this can be optimized with additional data, using a proof for a
   random evaluation at two points derived from the commitment and blob data)
 
-
-The signature is verified and `tx.origin` is calculated as follows:
+The hashes for signature verification and for determining the public transaction identifier are computed according to the [EIP-6493](./eip-6493.md) transaction signature scheme:
```
Comment (g11tech) [inline EIPS/eip-4844.md:165]:
```suggestion
The hashes for signature generation/verification and for determining the public transaction identifier are computed according to the [EIP-6493](./eip-6493.md) transaction signature scheme:
```

File: EIPS/eip-4844.md (hunk 2 above)
Comment (etan-status) [reply EIPS/eip-4844.md:177]:
```suggestion
def compute_tx_hash(signed_tx: SignedBlobTransaction) -> Root:
```

**Review by g11tech — CHANGES_REQUESTED** [review]
still under active discussion

**Comment by lightclient** [comment]
Are we still considering this? Or does the move away from SSZ render this path dead?

**Comment by etan-status** [comment]
For EIP-4844, there is nothing to gain by switching signature schemes. EIP-6493 is purely for SSZ transactions, which EIP-4844 no longer is.

**Pull request closed without merging**